include/envoy/network/dns.h

```diff
@@ -39,7 +39,8 @@ class DnsResolver {
    * @return if non-null, a handle that can be used to cancel the resolution.
    *         This is only valid until the invocation of callback or ~DnsResolver().
    */
-  virtual ActiveDnsQuery* resolve(const std::string& dns_name, ResolveCb callback) PURE;
+  virtual ActiveDnsQuery* resolve(const std::string& dns_name, uint32_t port,
```

Comment:
This interface needs some documenting. I think it might be a little confusing to the reader. In general, an A record is independent of port, so I'm not sure it makes sense to expect the caller to supply a port in the non-SRV case, at least conceptually.

Is the rationale that we're eventually providing an Address::Ipv4Instance which is qualified with port? If this is the case, I think it might make sense to have a variant of the Address class that is also independent of port, similar to the C level distinction between in_addr and sockaddr_in structs. What do you think @mattklein123? I think @jamessynge was also after something like this.